src/main/java/teammates/logic/api/EmailGenerator.java

```diff
@@ -469,198 +450,203 @@ private EmailWrapper generateSessionLinksRecoveryEmailForExistingStudent(String
+        return generateFeedbackSessionEmailBases(course, session, students, instructors, instructorsToNotify,
+                template, EmailType.FEEDBACK_PUBLISHED, FEEDBACK_ACTION_VIEW);
     }
 
     /**
      * Generates the feedback session published emails for the given {@code session}.
      */
     public List<EmailWrapper> generateFeedbackSessionUnpublishedEmails(FeedbackSessionAttributes session) {
-
-        String template = EmailTemplates.USER_FEEDBACK_SESSION_UNPUBLISHED;
-
         CourseAttributes course = coursesLogic.getCourse(session.getCourseId());
-        boolean isEmailNeeded = fsLogic.isFeedbackSessionViewableToStudents(session);
-        List<InstructorAttributes> instructors = isEmailNeeded
-                                                 ? instructorsLogic.getInstructorsForCourse(session.getCourseId())
-                                                 : new ArrayList<>();
-        List<StudentAttributes> students = isEmailNeeded
-                                           ? studentsLogic.getStudentsForCourse(session.getCourseId())
-                                           : new ArrayList<>();
-
-        return generateFeedbackSessionEmailBases(course, session, students, instructors, template,
-                EmailType.FEEDBACK_UNPUBLISHED);
-    }
+        boolean isEmailNeededForStudents = fsLogic.isFeedbackSessionViewableToUserType(session, false);
+        boolean isEmailNeededForInstructors = fsLogic.isFeedbackSessionViewableToUserType(session, true);
+        List<InstructorAttributes> instructorsToNotify = isEmailNeededForStudents
+                ? instructorsLogic.getCoOwnersForCourse(session.getCourseId())
+                : new ArrayList<>();
+        List<StudentAttributes> students = isEmailNeededForStudents
+                ? studentsLogic.getStudentsForCourse(session.getCourseId())
+                : new ArrayList<>();
+        List<InstructorAttributes> instructors = isEmailNeededForInstructors
+                ? instructorsLogic.getInstructorsForCourse(session.getCourseId())
+                : new ArrayList<>();
 
-    private List<EmailWrapper> generateFeedbackSessionEmailBases(
-            CourseAttributes course, FeedbackSessionAttributes session, List<StudentAttributes> students,
-            List<InstructorAttributes> instructors, String template, EmailType type) {
-        String additionalContactInformation = getAdditionalContactInformationFragment(course);
-        return generateFeedbackSessionEmailBases(course, session, students, instructors, template, type,
-                FEEDBACK_ACTION_SUBMIT_EDIT_OR_VIEW, additionalContactInformation);
+        String template = EmailTemplates.USER_FEEDBACK_SESSION_UNPUBLISHED;
+        return generateFeedbackSessionEmailBases(course, session, students, instructors, instructorsToNotify, template,
+                EmailType.FEEDBACK_UNPUBLISHED, FEEDBACK_ACTION_SUBMIT_EDIT_OR_VIEW);
```

Comment:
Instead of `FEEDBACK_ACTION_SUBMIT_EDIT_OR_VIEW`, maybe consider an empty string instead? There is no `feedbackAction` key in the email template, so doesn't really make a difference either way, but it doesn't seem appropriate because the feedback session might be closed when the session is unpublished.

Reply:
This is harmless and redundant at best, as how it was done before. In fact, if you look at `generateFeedbackSessionEmailBaseFor*`, there are a handful of similarly redundant replacements.